packages/moleculer-db-adapter-mongoose/src/index.js

```diff
@@ -78,38 +77,41 @@ class MongooseDbAdapter {
 			} else {
 				conn = mongoose.connect(this.uri, this.opts);
 			}
+	
 		} else if (this.schema) {
-			conn = mongoose.createConnection(this.uri, this.opts);
-			this.model = conn.model(this.modelName, this.schema);
-		}
-
+			
+			conn = new Promise(resolve =>{
+				const 	c = mongoose.createConnection(this.uri, this.opts);
+				this.model = c.model(this.modelName, this.schema);
+				resolve(c);
+			});}
+
+			
 		return conn.then(_result => {
 			const result = _result || conn;
-			this.conn = conn;
-
-			if (mongoose.connection.readyState != mongoose.connection.states.connected) {
-				throw new MoleculerError(
-					`MongoDB connection failed . Status is "${
-						mongoose.connection.states[mongoose.connection._readyState]
-					}"`
-				);
-			}
+			this.conn = this.isPromise(conn)? result: conn;
+		
+
+			if(this.model)
+				this.model = _result.model(this.model["modelName"],this.model["schema"]);
+
 
 			if (result.connection)
 				this.db = result.connection.db;
 			else
 				this.db = result.db;
 
-			if (!this.db) {
```

Comment:
Because there is no scenario where the DB can be null, if this happens because the connection was not carried out, it will not be a promise generating an error before reaching these validations.

Reply:
ok